drivers/storage/portworx/component/telemetry.go

```diff
@@ -178,12 +183,18 @@ func (t *telemetry) reconcileCCMGo(
 	cluster *corev1.StorageCluster,
 	ownerRef *metav1.OwnerReference,
 ) error {
-	if err := t.reconcileCCMGoRolesAndRoleBindings(cluster, ownerRef); err != nil {
+	// Remove old ccm components first if exist
+	if err := k8sutil.DeleteConfigMap(t.k8sClient, TelemetryCCMProxyConfigMapName, cluster.Namespace, *ownerRef); err != nil {
 		return err
 	}
-	if cluster.Status.ClusterUID == "" {
-		logrus.Warn("clusterUID is empty, wait for it to fill telemetry proxy config")
-		return nil
+	if err := t.createCCMGoSCCComponents(cluster, ownerRef); err != nil {
+		return err
+	}
+	if err := t.deployMetricsCollector(cluster, ownerRef); err != nil {
```

Comment:
metrics collector uses ccm roles and rolebindings, so this should be moved to after those are created.

Reply:
The deploy function creates all objects needed except the scc related ones

Reply:
what's the reason to share clusterRole but not share role.

Reply:
looks like the new ccm does not have a role, perhaps it's correct, we shall test it on various environment to see if any permission is missing.